plugins/database/mongodb/connection_producer.go

```diff
@@ -201,7 +206,37 @@ func parseMongoURL(rawURL string) (*mgo.DialInfo, error) {
 			}
 			if ssl {
 				info.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
-					return tls.Dial("tcp", addr.String(), &tls.Config{})
+					tlsConfig := &tls.Config{}
+					if sslCert != "" && sslKey != "" && sslCA != "" {
+						clientCertPEM, err := ioutil.ReadFile(sslCert);
```

Comment:
The API server will not have access to local resources.  You must consume the PEM encoded string directly.

Reply:
Thanks, I totally forgot about that. I have changed the code to consume as strings instead